### ISSUE TITLE
Sub-menus in benefits mega menu don't close with spacebar input

### ISSUE THREAD

**Issue by davidmpickett** [department-of-veterans-affairs/va.gov-cms#18368]
## Describe the defect
Keyboard users can't easily close submenus, so they have to navigate through the whole submenu to get to the next menu item. A mouse user can click open the next menu immediately.

## To Reproduce
1. Go to https://www.va.gov/
2. Tab to the "VA Benefits and Health Care" menu item
3. Notice that you can open and close this menu item with space bar.
4. Tab through the menu items in the "VA Benefits and Health Care" menu
5. Open one of the menu items (e.g. "Health Care") with space bar
6. Notice that you can't close this sub menu by pressing space bar. The only way to get to the next menu item (e.g. "Disability"" is to tab through every item in the sub menu.

## AC / Expected behavior
Keyboard users should be able to close sub menus.

## Screenshots
![Screenshot 2024-06-18 at 6 47 40 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/fd9de806-ffff-4837-a0aa-c367e78883b6)

## Desktop (please complete the following information if relevant, or delete)
 - OS: masOS 14.5 
 - Browser: Chrome 126.0.6478.62
